src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -400,6 +411,23 @@ public HttpResponse doFinishLogin(StaplerRequest request)
         return HttpResponses.redirectToContextRoot();   // referer should be always there, but be defensive
     }
 
+    private void storeCredentials(String accessToken, GithubAuthenticationToken auth) throws IOException {
+        CredentialsStore credentialsStore = CredentialsProvider.lookupStores(Jenkins.getInstance()).iterator().next();
+        Credentials github = Iterables.find(credentialsStore.getCredentials(Domain.global()), new Predicate<Credentials>() {
```

Comment:
No. no no no. That forces the credentials provider to fully instantiate the credentials instances which will log a password access against external provider implementations.